alice/middleware/logger.go

```diff
@@ -22,35 +27,38 @@ type logger struct {
 	logRequests bool
 }
 
-type loggingResponseWriter struct {
-	http.ResponseWriter
-	StatusCode int
-}
-
-func (l *loggingResponseWriter) WriteHeader(code int) {
-	l.StatusCode = code
-	l.ResponseWriter.WriteHeader(code)
-}
-
 // NewLogrusLogger allows you to setup a base entry to use for logging
 func NewLogrusLogger(baseEntry *logrus.Entry, logRequests bool) Logger {
 	return &logger{entry: baseEntry, logRequests: logRequests}
 }
 
 func (l *logger) Log(h http.Handler) http.Handler {
 	return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
-		entry := l.entry.WithFields(logrus.Fields{
+		loggingResponseWriter := &lrw.LoggingResponseWriter{w, http.StatusOK, nil}
+
+		fields := logrus.Fields{
 			logFieldRequestID: getRequestIDFromContext(r.Context()),
 			logFieldUserID:    getInsecureUserIDFromContext(r.Context()),
-		})
+			logFieldMethod:    r.Method,
+			logFieldPath:      r.URL.String(),
+		}
+
+		entry := l.entry.WithFields(fields)
+
 		// add logger to the context
 		ctx := context.WithValue(r.Context(), contextKeyLogger, entry)
 		r = r.WithContext(ctx)
 
-		lrw := &loggingResponseWriter{w, http.StatusOK}
-		h.ServeHTTP(lrw, r)
+		h.ServeHTTP(loggingResponseWriter, r)
+
+		fields[logFieldStatusCode] = loggingResponseWriter.StatusCode
+
+		if loggingResponseWriter.InnerError != nil {
+			fields[logFieldError] = loggingResponseWriter.InnerError
```

Comment:
Don't you need to setup a new entry with these fields?

Reply:
In my testing, I did not. It added the fields to the entry. Unless there is something around using the entry more than once where I need to add them again. I would assume that since the fields are just a map and those are pointers it is safe to use the same fields for the entry

Reply:
entry must hold a reference to the fields object..... do the regular logs get the new fields after and error?